build.sbt

```diff
@@ -167,45 +203,46 @@ lazy val `play-json` = crossProject(JVMPlatform, JSPlatform)
           .map {
             i =>
               def commaSeparated(s: Int => String)   = 1.to(i).map(s).mkString(", ")
-              def newlineSeparated(s: Int => String) = 1.to(i).map(s).mkString("\n")
+              def newlineSeparated(s: Int => String) = 1.to(i).map(s).mkString("\n        ")
               val writerTypes                        = commaSeparated(j => s"T$j: Writes")
               val readerTypes                        = commaSeparated(j => s"T$j: Reads")
               val typeTuple                          = commaSeparated(j => s"T$j")
               val written                            = commaSeparated(j => s"implicitly[Writes[T$j]].writes(x._$j)")
               val readValues                         = commaSeparated(j => s"t$j")
               val readGenerators                     = newlineSeparated(j => s"t$j <- implicitly[Reads[T$j]].reads(arr(${j - 1}))")
 
-              (s"""
-          implicit def Tuple${i}W[$writerTypes]: Writes[Tuple${i}[$typeTuple]] = Writes[Tuple${i}[$typeTuple]](
-            x => JsArray(Array($written))
-          )
-          """, s"""
-          implicit def Tuple${i}R[$readerTypes]: Reads[Tuple${i}[$typeTuple]] = Reads[Tuple${i}[$typeTuple]]{
-            case JsArray(arr) if arr.size == $i =>
-              for{
-                $readGenerators
-              } yield Tuple$i($readValues)
-
-            case _ =>
-              JsError(Seq(JsPath() -> Seq(JsonValidationError("Expected array of $i elements"))))
-          }
-        """)
+              val writes =
```

Comment:
Maybe with the use of the margins it would be equally readable as a single expression.

But, more generally, I lean towards trying to make statements/expressions less multi-line, like so.